examples/bidder/http_bidder_test.cpp

```diff
@@ -95,49 +103,63 @@ int main(int argc, char *argv[]) {
         LOG(error) << e.what();
         return 0;
     }
-    exchange_handler<DSL::GenericDSL<>> bid_handler(std::chrono::milliseconds(config.data().timeout));
+    
+
+    using bid_handler_type = exchange_handler<DSL::GenericDSL<>, vanilla::VanillaRequest>;
+    using BidRequest = bid_handler_type::auction_request_type;
+    
+    bid_handler_type bid_handler(std::chrono::milliseconds(config.data().timeout));
+       
+    enum class request_user_data {USER_DATA, NO_BID, AUCTION_ASYNC, COUNT};
+    using decision_codes_type = request_user_data;
+    using decision_exchange_type = vanilla::decision_exchange::decision_exchange<decision_codes_type, http::server::reply&, BidRequest&>;
+    
+    auto request_user_data_f = [&bid_handler, &config](http::server::reply &reply, BidRequest & bid_request) -> bool {
+        using kv_type = vanilla::client::empty_key_value_client;
+        thread_local kv_type kv_client;
+        bool is_matched_user = bid_request.user_info.user_id.length();
+        if (!is_matched_user) {
+            return true; // bid unmatched
+        }
+        if (!kv_client.connected()) {
+            kv_client.connect(config.data().key_value_host, config.data().key_value_port);
+        }
+        kv_client.request(bid_request.user_info.user_id, bid_request.user_info.user_data);
+        return true;
+    };
+    auto no_bid_f = [&bid_handler, &config](http::server::reply &reply, BidRequest & bid_request) -> bool {
+        reply << http::server::reply::flush("");
+    };
+    auto auction_async_f = [&bid_handler](http::server::reply &reply, BidRequest & bid_request) -> bool {
+        return bid_handler.handle_auction_async(reply, bid_request);
+    };
+    const decision_exchange_type::decision_tree_type decision_tree = {{
```

Comment:
The sample code below is more readable because it's very compact, and yes I agree it's hard to see the output routes, so formatting definitely helps to improve readability even with enums. 

```
decision_tree_manager<3> tm = decision_tree<3>
{{
    { 0 , { action_1, -1 , 1}  }, //true->EXIT , false->index=1
    { 1 , { action_2, -1 , 2}  }, //true->EXIT , false->index=2
    { 2 , { action_3, -1 , -1} }  //EXIT
}} ;
```